src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -33,6 +33,7 @@
 import com.openshift.restclient.http.IHttpConstants;
 import com.openshift.restclient.utils.SSLUtils;
 
+import okhttp3.Authenticator;
```

Comment:
@fbricon @adietish This would be the first place where we publicly expose okhttp to consumers of this client library.  I would prefer we had some sort of abstraction though I don't know how practical it is to do so.  Thoughts on allowing the authenticator to go public?  I have reservations.